x-pack/plugin/core/build.gradle

```diff
@@ -39,6 +39,10 @@ dependencies {
     // security deps
     shadow 'com.unboundid:unboundid-ldapsdk:3.2.0'
     shadow project(path: ':modules:transport-netty4', configuration: 'runtime')
+    shadow(project(path: ':plugins:transport-nio', configuration: 'runtime')) {
+        // TODO: This should not be necessary if core is compile only
```

Comment:
I would expand this comment a bit. Something like:
```
// TODO: core exclusion should not be necessary, since it is a transitive dep of all plugins
```

Also, I wonder if this is due to how we add server as a `compile` dep, but this is shadow here? I wonder if @nik9000's upcoming changes to add `bundle` will make this better?